ksp_plugin/flight_plan_optimization_driver.hpp

```diff
@@ -0,0 +1,74 @@
+#pragma once
+
+#include <memory>
+
+#include "absl/synchronization/mutex.h"
+#include "base/jthread.hpp"
+#include "base/not_null.hpp"
+#include "ksp_plugin/celestial.hpp"
+#include "ksp_plugin/flight_plan.hpp"
+#include "ksp_plugin/flight_plan_optimizer.hpp"
+#include "quantities/named_quantities.hpp"
+#include "quantities/quantities.hpp"
+
+namespace principia {
+namespace ksp_plugin {
+namespace _flight_plan_optimization_driver {
+namespace internal {
+
+using namespace principia::base::_jthread;
+using namespace principia::base::_not_null;
+using namespace principia::ksp_plugin::_celestial;
+using namespace principia::ksp_plugin::_flight_plan;
+using namespace principia::ksp_plugin::_flight_plan_optimizer;
+using namespace principia::quantities::_named_quantities;
+using namespace principia::quantities::_quantities;
+
+class FlightPlanOptimizationDriver {
```

Comment:
This needs a way to determine whether an optimization is in progress, or whether it is done.